### PR TITLE
Refactor appregistry and datastore package

### PR DESCRIPTION
**Pull request by tkashem**
Refactor appregistry and datastore package to pave the way for future work on YAML parsing of manifest blob.

- [appregistry] fix typo in "unmarshaller.go"
- [appregistry] move operator manifest type(s) out of appregistry
- [datastore] Move YAML parsing logic into datastore package 
- [datastore] rename datastore types

### CODE REVIEW

File: pkg/datastore/manifest.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package datastore
+
+// Manifest encapsulates operator manifest data
```
Comment (aravindhp) [inline pkg/datastore/manifest.go]: Please add a `.` at the end of all sentences here and in other places.

File: pkg/datastore/metadata.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package datastore
+
+import (
+	"fmt"
+)
+
+// OperatorMetadata encapsulates operator metadata and manifest assocated with a package
```
Comment (aravindhp) [inline pkg/datastore/metadata.go]: Comment should be 80 column formatted. Please fix here and in other places.

File: pkg/datastore/metadata.go [hunk 3]
```diff
@@ -4,24 +4,35 @@ import (
...
 	Namespace string
 
-	// Repository is the repository name for the specified package in app registry
+	// The repository inside the given namespace where operator manifest(s)
+	// is located.
 	Repository string
 
-	// Release represents the release or version number of the given package
+	// Release represents the release or version number of the given
+	// operator manifest(s)
```
Comment (aravindhp) [inline pkg/datastore/metadata.go]: Missing period at the end

File: pkg/datastore/metadata.go [hunk 4]
```diff
@@ -4,24 +4,35 @@ import (
 	"fmt"
 )
 
-// OperatorMetadata encapsulates operator metadata and manifest assocated with a package
+// OperatorMetadata encapsulates operator metadata and manifest
```
Comment (aravindhp) [inline pkg/datastore/metadata.go]: This and other formatting fixes should have been done in the `[appregistry] move operator manifest type(s) out of appregistry` but I won't insist on it.

File: pkg/datastore/mock_unmarshaller.go [hunk 5]
```diff
@@ -1,12 +1,11 @@
 // Code generated by MockGen. DO NOT EDIT.
```
Comment (aravindhp) [inline pkg/datastore/mock_unmarshaller.go]: Please add a task to generate the appregistry mock files when the unit tests are run like you did for OperatorSource. We can then remove them from the repo.

File: pkg/appregistry/unmarshaller.go [hunk 6]
```diff
@@ -20,12 +20,12 @@ type Data struct {
 	// CSVs is the list of CSV associated with a package
 	CSVs string `yaml:"clusterServiceVersions"`
 
-	// Packages is the list of channles associated with a package
+	// Packages is the list of channels associated with a package
 	Packages string `yaml:"packages"`
 }
 
 type blobUnmarshaller interface {
```
[the remark this answers is not in the document]
Comment (tkashem) [reply pkg/appregistry/unmarshaller.go]: Both `unmarshaller` and `unmarshaler` are correct. I checked Go runtime library and it uses `unmarshaler` - https://golang.org/src/encoding/json/decode.go?s=4757:4816#L110.

To be consistent, I will change it to `unmarshaller`.

File: pkg/datastore/datastore_test.go [hunk 7]
```diff
@@ -1,26 +1,36 @@
...
 	expected := []string{"foo/1", "bar/2", "braz/3"}
 
-	packages := []*appregistry.OperatorMetadata{
-		&appregistry.OperatorMetadata{Namespace: "foo", Repository: "1"},
-		&appregistry.OperatorMetadata{Namespace: "bar", Repository: "2"},
-		&appregistry.OperatorMetadata{Namespace: "braz", Repository: "3"},
+	packages := []*OperatorMetadata{
+		helperNewOperatorMetadata("foo", "1"),
+		helperNewOperatorMetadata("bar", "2"),
+		helperNewOperatorMetadata("braz", "3"),
```
Comment (SamiSousa) [inline pkg/datastore/datastore_test.go]: Minor nit: Any reason behind `braz` here? I would have expected `foobar`

Comment (tkashem) [reply pkg/datastore/datastore_test.go]: Actually, this is a typo. It should be `baz`. 
From Wikipedia - `foo, bar, baz, qux, quux, quuz, corge, grault, garply, waldo, fred, plugh, xyzzy, and thud.[1][3] Wibble, wobble, wubble, and flob are also used in the UK.` :)

You are welcome to fix the typo in a separate PR :)

File: pkg/datastore/mock_unmarshaller.go (hunk 5 above)
Comment (tkashem) [reply pkg/datastore/mock_unmarshaller.go]: Done.

File: pkg/datastore/datastore_test.go (hunk 7 above)
Comment (SamiSousa) [reply pkg/datastore/datastore_test.go]: Actually, I like the sound of `braz` over something like `garply`

File: pkg/appregistry/unmarshaller.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (aravindhp) [reply pkg/appregistry/unmarshaller.go]: Yes please be consistent. I know Java uses unmarshaller.

**Pull request merged**